course/api/views.py

```diff
@@ -32,13 +35,16 @@
     IsCourseAdminOrUserObjIsSelf,
     OnlyEnrolledStudentOrCourseStaffPermission,
 )
+from ..forms import (
+    CourseInstanceForm,
+)
 from .serializers import *
 from .full_serializers import *
 
 
 class CourseViewSet(ListSerializerMixin,
                     CourseResourceMixin,
-                    viewsets.ReadOnlyModelViewSet):
+                    viewsets.ModelViewSet):
```

Comment:
Even worse, is any logged-in user allowed to delete since the course list has had very lenient permission classes? (FYI, the default is set in settings.py, REST_FRAMEWORK['DEFAULT_PERMISSION_CLASSES'])

Reply:
Permission classes should now be used appropriately, I hope.